test/jdk/java/nio/channels/FileChannel/TransferToChannel.java

```diff
@@ -124,14 +152,42 @@ static void transferFileToTrustedChannel() throws Exception {
         out.close();
     }
 
-    static void generateBigFile(File file) throws Exception {
+    private static void transferFileDirectly() throws Exception {
+        outFile.delete();
+        FileOutputStream fos = new FileOutputStream(outFile);
+        FileChannel out = fos.getChannel();
+
+        final long size = in.size();
+        final long position = RAND.nextInt((int)size - MIN_DIRECT_TRANSFER_SIZE);
+        assert out.position() == 0;
+        long pos = position;
+        while (pos < size) {
+            long bytesTransferred = in.transferTo(pos, Long.MAX_VALUE, out);
+            if (bytesTransferred >= 0)
+                pos += bytesTransferred;
+            else
+                throw new Exception("transfer failed");
```

Comment:
Fixed by commit c72b4b58f57d998423fb94b17ac950185109b933.

> I would probably have been tempted to convert this test to leverage TestNG as I think it would streamline the test a bit (or safe this for a future date)

I prefer to leave the TestNG conversion to another issue TBD.